src/SFA.DAS.EAS.DbMaintenance.WebJob/Jobs/PaymentIntegrityCheckerJob.cs

```diff
@@ -0,0 +1,74 @@
+﻿using SFA.DAS.EAS.Domain.Data.Repositories;
+using SFA.DAS.EAS.Domain.Interfaces;
+using SFA.DAS.EAS.Domain.Models.Payments;
+using SFA.DAS.NLog.Logger;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EAS.DbMaintenance.WebJob.Jobs
+{
+    public class PaymentIntegrityCheckerJob : IJob
+    {
+        private readonly IPaymentService _paymentService;
+        private readonly IEmployerAccountRepository _employerAccountRepository;
+        private readonly IDasLevyRepository _levyRepository;
+        private readonly ILog _logger;
+
+        public PaymentIntegrityCheckerJob(
+            IPaymentService paymentService,
+            IEmployerAccountRepository employerAccountRepository,
+            IDasLevyRepository levyRepository,
+            ILog logger)
+        {
+            _paymentService = paymentService;
+            _employerAccountRepository = employerAccountRepository;
+            _levyRepository = levyRepository;
+            _logger = logger;
+        }
+
+        public async Task Run()
+        {
+            var accounts = await _employerAccountRepository.GetAllAccounts();
+
+            var periodEnds = await _levyRepository.GetAllPeriodEnds();
+
+            foreach (var periodEnd in periodEnds)
+            {
+                foreach (var account in accounts)
+                {
+                    var expectedPayments =
+                        (IEnumerable<Payment>)await _paymentService.GetAccountPayments(periodEnd.Id,
```

Comment:
Is the cast required? The returned ICollection<T> is already an IEnumerable<T> (PaymentDetail is a subclass of Payment)
The return type can't be null (IEnumerable derived return types are not supposed to be null) and in this case the actual called code will never return null,  so the subsequent Elvis operators starting at L 49 aren't required.